pkg/loghttp/entry.go

```diff
@@ -52,26 +54,46 @@ func (e *Entry) UnmarshalJSON(data []byte) error {
 				return
 			}
 			e.Line = v
-		case 2: // labels
+		case 2: // structured metadata
 			if t != jsonparser.Object {
 				parseError = jsonparser.MalformedObjectError
 				return
 			}
+
 			var structuredMetadata labels.Labels
+			var parsed labels.Labels
 			if err := jsonparser.ObjectEach(value, func(key []byte, value []byte, dataType jsonparser.ValueType, _ int) error {
-				if dataType != jsonparser.String {
-					return jsonparser.MalformedStringError
+				if dataType == jsonparser.Object {
```

Comment:
can you please leave a comment saying we parse both push request and query response json here to make it clear for others?

Reply:
Done